File: 프로그래머스/lv0/120861. 캐릭터의 좌표/캐릭터의 좌표.py
def solution(keyinput, board):
    answer = [0, 0]
    width = [-(board[0]//2), board[0]//2]
    height = [-(board[1]//2), board[1]//2]
    
    for i in keyinput:
        if i == "left" and answer[0] > width[0]:
            answer[0] -= 1
        elif i == "right" and answer[0] < width[1]:
            answer[0] += 1
        elif i == "up" and answer[1] < height[1]:
            answer[1] += 1
        elif i == "down" and answer[1] > height[0]:
            answer[1] -= 1

    return answer

File: 프로그래머스/0/120861. 캐릭터의 좌표/캐릭터의 좌표.java
import java.util.Arrays;

class Solution {
    public int[] solution(String[] keyInput, int[] board) {
        int[] answer = {0, 0};
        int width = board[0] / 2;
        int height = board[1] / 2;

        for (String key : keyInput) {
            if ("left".equals(key) && width >= Math.abs(answer[0] - 1)) {
                answer[0] -= 1;
            } else if ("right".equals(key) && width >= Math.abs(answer[0] + 1)) {
                answer[0] += 1;
            } else if ("up".equals(key) && height >= Math.abs(answer[1] + 1)) {
                answer[1] += 1;
            } else if ("down".equals(key) && height >= Math.abs(answer[1] - 1)) {
                answer[1] -= 1;
            }
        }
        return answer;
    }

    public static void main(String[] args) {
        Solution T = new Solution();
        String[] keyInput = {"left", "left", "left", "right"};
        int[] board = {3, 3};

        System.out.println(Arrays.toString(T.solution(keyInput, board)));
    }
}
